lib/oxrdf/src/triple.rs

```diff
@@ -780,17 +793,86 @@ impl<'a> From<TermRef<'a>> for Term {
 /// # Result::<_,oxrdf::IriParseError>::Ok(())
 /// ```
 #[derive(Eq, PartialEq, Debug, Clone, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct Triple {
     /// The [subject](https://www.w3.org/TR/rdf11-concepts/#dfn-subject) of this triple.
     pub subject: Subject,
 
     /// The [predicate](https://www.w3.org/TR/rdf11-concepts/#dfn-predicate) of this triple.
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            serialize_with = "serialize_predicate",
+            deserialize_with = "deserialize_predicate"
+        )
+    )]
     pub predicate: NamedNode,
 
     /// The [object](https://www.w3.org/TR/rdf11-concepts/#dfn-object) of this triple.
     pub object: Term,
 }
 
+#[cfg(feature = "serde")]
+fn serialize_predicate<S>(node: &NamedNode, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    // Serialize the NamedNode as { "type": "uri", "value": "..." }
+    let mut state = serializer.serialize_struct("NamedNode", 2)?;
+    state.serialize_field("type", "uri")?;
+    state.serialize_field("value", &node.as_str())?;
+    state.end()
+}
+
+#[cfg(feature = "serde")]
+fn deserialize_predicate<'de, D>(deserializer: D) -> Result<NamedNode, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct PredicateVisitor;
```

Comment:
what about deserializing into a `Term` then failing if the term is not a `NamedNode`? Will be shorter and I think as correct.

Reply:
Similarly to https://github.com/oxigraph/oxigraph/pull/1185#discussion_r1960496168, only concern would be performance-related.

Reply:
There appears to be a non-negligible performance regression when doing this

```
deserialize_triple      time:   [512.28 ns 513.97 ns 515.64 ns]
                        change: [+9.7608% +12.148% +14.066%] (p = 0.00 < 0.05)
                        Performance has regressed.
```

Reply:
However, having run multiple tests on my machine - approximately half of that regression may be noise